openedx/core/djangoapps/user_api/learner_skill_levels/api.py

```diff
@@ -0,0 +1,120 @@
+"""
+APIs for learner skill levels.
+"""
+from opaque_keys.edx.locator import CourseLocator
+
+from lms.djangoapps.grades.models import PersistentCourseGrade  # lint-amnesty, pylint: disable=unused-import
+from openedx.core.djangoapps.catalog.utils import get_course_data
+
+from .constants import LEVEL_TYPE_SCORE_MAPPING
+
+
+def _calculate_user_skill_score(all_skills):
+    """
+    Calculates user skill score to see where the user falls in a certain job category.
+    """
+    sum_of_skills = sum(all_skills.values())
+    skills_count = len(all_skills)
+    # sum of skills score in the category/ 3*no. of skills in category
+    if skills_count:
+        return round(sum_of_skills / (3 * skills_count), 1)
+    return 0.0
+
+
+def _get_category_skill_score(category_skills, learner_skill_score):
+    """
+    Takes each skill item in dict and appends its score to it.
+    For a skill that doesn't exist in learner's skills set, appends None as score.
+    """
+    return [
+        dict(skill, **{'score': learner_skill_score.get(skill['name'], None)})
+        for skill in category_skills
+    ]
```

Comment:
Why we are generating new dicts here? can we just loop over the skills and add new key value as skill['score'] = learner_skill_score.get(skill['name'])

Reply:
Opps, this was initially written when we were removing skills that were not part of learner's skills. Generating a new dict made more sense back then as we were running the loop over 100 users.